src/baldr/graphreader.cc

```diff
@@ -104,6 +104,67 @@ GraphReader::get_extract_instance(const boost::property_tree::ptree& pt) {
   return tile_extract;
 }
 
+// ----------------------------------------------------------------------------
+// FlatTileCache implementation
+// ----------------------------------------------------------------------------
+
+// Constructor.
+FlatTileCache::FlatTileCache(size_t max_size) : cache_size_(0), max_cache_size_(max_size) {
```

Comment:
instead of hashing we use a vector to store a spot for all the tile pointers that could ever exist. this is slightly faster than hashing and the indirection in unordered_map but at the cost of a around 8 megabytes of ram iirc

Reply:
yeah true, its 8.5MB for all levels:

(1036800 (level 3 tiles) + 1036800 (level 2 tiles) + 64800 (level 1)  + 4050 (level 0)) * 4 bytes

Do you think the extra work would be worth the 500k of RAM? imho modern devices are past the point of worrying about those types of ram requirements. I do agree though that we should have a config that lets you select simple if you dont want to pay the 8MB toll on mobile devices. I'll add an issue for that

Reply:
not worth worrying about 500K RAM, but agree on adding an issue to add a config setting to allow simple cache if desired.

Reply:
ive just pushed a commit to enable this. no issue needed!